File: arrays/return_sum_1_d_array.py
# Given an array nums.

# We define a running sum of an array as runningSum[i] = sum(nums[0]…nums[i]).

# Return the running sum of nums.

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        for i in range(len(nums) - 1):
            nums[ i+1 ] += nums[i]
        
        return nums

File: arrays/return_sum_1_d_array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    let sum = 0;
    const newArray = nums.map(item => {
        sum = item + sum;
        return sum;
    });

    return newArray
};



console.log(runningSum([1,2,3,4,5])) // [1,3,6,10,15]
